src/jit/codegenxarch.cpp

```diff
@@ -3276,11 +3276,13 @@ void CodeGen::genStructPutArgUnroll(GenTreePutArgStk* putArgNode)
 
     assert(src->isContained());
 
-    assert(src->gtOper == GT_OBJ);
+    assert(src->gtOper == GT_OBJ || src->gtOper == GT_LCL_VAR);
+    
+    GenTree* op1 = src->gtOper == GT_OBJ ? src->gtOp.gtOp1 : src;
```

Comment:
The issue here is that op1 of a `GT_OBJ` is an address, whereas `GT_LCL_VAR` is a value. So, two cases are needed for the lclVar:
- If it's in a register, then you need a different code path that moves the value from the register to the stack (and maybe it shouldn't wind up in the unroll case).
- If it's on the stack, then you need a lclVar address.